contracts/amm/contract/src/lib.rs

```diff
@@ -42,9 +47,76 @@ impl Hash for UnorderedTokenPair {
 }
 
 pub struct AmmContract {
+    pub exec_state: ExecutionState,
     contract_name: ContractName,
     state: AmmState,
-    caller: Identity,
+}
+
+impl CallerCallee for AmmContract {
+    fn caller(&self) -> &Identity {
+        &self.exec_state.caller
+    }
+    fn callees_blobs(&self) -> CalleeBlob {
+        CalleeBlob(self.exec_state.callees_blobs.borrow())
+    }
+    fn mut_callees_blobs(&self) -> MutCalleeBlob {
+        MutCalleeBlob(self.exec_state.callees_blobs.borrow_mut())
+    }
+}
+
+impl<T: PartialEq + 'static> ConnectToContract<T> for AmmContract
+where
+    StructuredBlob<T>: TryFrom<Blob>,
+    <StructuredBlob<T> as TryFrom<Blob>>::Error: std::fmt::Debug,
+{
+    fn connect_to(&self, contract_name: String) -> Contract<T> {
+        Contract::new(&self.exec_state.callees_blobs, contract_name)
+    }
+}
+
+struct IErc20<'a>(Contract<'a, ERC20Action>);
+impl<'a> Deref for IErc20<'a> {
+    type Target = Contract<'a, ERC20Action>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
```

Comment:
I find this part and the next very confusing. I imagine myself as the developper of the AMM trying to use the hyle SDK, It's not clear how I should use it, and why I have to write the boilerplate. Not very handy to use IMHO